File: python/binary_search.py
import math


def search(num, array):
    lowerIndex = 0
    higherIndex = len(array) - 1

    while lowerIndex <= higherIndex:
        med = math.floor((lowerIndex + higherIndex) / 2)
        pivot = array[med]
        if pivot == num:
            return med
        elif num < pivot:
            higherIndex = med-1
        elif num > pivot:
            lowerIndex = med+1

    return -1


def main():
    array = [1, 2, 10, 12, 17, 18, 19, 20, 23, 24, 25, 26, 27, 29, 30, 31, 33, 34, 36, 37]
    print(search(30, array))


main()

File: java/binarysearch/BinarySearch.java
package binarysearch;


import java.util.List;

public class BinarySearch {

    public int search(int number, List<Integer> myList) {
        int lowerIndex = 0;
        int higherIndex = myList.size() - 1;

        while (lowerIndex <= higherIndex) {
            int pivot = (lowerIndex + higherIndex) / 2;
            if (number == myList.get(pivot)) return pivot;
            if (number < myList.get(pivot)) {
                higherIndex = pivot - 1;
            } else {
                lowerIndex = pivot + 1;
            }
        }
        return -1;
    }
}